common/bitops.h

```diff
@@ -0,0 +1,47 @@
+#ifndef COMMON_BITOPS_H
+#define COMMON_BITOPS_H
+
+#include <cassert>
+#include <climits>
+
+namespace sim {
```

Comment:
Please, rename namespace to 'simulator' or change namespace name to 'sim' in memory folder